monai/apps/reconstruction/transforms/dictionary.py

```diff
@@ -57,15 +58,25 @@ def __call__(self, data: Mapping[Hashable, NdarrayOrTensor]) -> dict[Hashable, T
         Returns:
             the new data dictionary
         """
+
         d = dict(data)
+
+        if isinstance(d[self.meta_key], MetaTensor):
+            # meta tensor
+            meta = d[self.meta_key].meta
+        else:
+            # meta dict
+            meta = d[self.meta_key]
```

Comment:
```suggestion
        meta: dict[str, Any]
        if isinstance(d[self.meta_key], MetaTensor):
            # meta tensor
            meta = d[self.meta_key].meta  # type: ignore
        else:
            # meta dict
            meta = d[self.meta_key]
```

This might be needed to get around the type issue.